src/ln/channelmonitor.rs

```diff
@@ -1123,36 +1185,41 @@ impl ChannelMonitor {
 					htlc_success_tx.input[0].witness[2].push(SigHashType::All as u8);
 
 					htlc_success_tx.input[0].witness.push(payment_preimage.to_vec());
-					htlc_success_tx.input[0].witness.push(chan_utils::get_htlc_redeemscript_with_explicit_keys(htlc, &local_tx.a_htlc_key, &local_tx.b_htlc_key, &local_tx.revocation_key).into_bytes());
+					let redeemscript = chan_utils::get_htlc_redeemscript_with_explicit_keys(htlc, &local_tx.a_htlc_key, &local_tx.b_htlc_key, &local_tx.revocation_key);
+					htlc_success_tx.input[0].witness.push(redeemscript.clone().into_bytes());
 
+					if let Some(ref per_commitment_point) = *per_commitment_point {
+						spendable_outputs.push((Some(per_commitment_point.clone()), redeemscript, BitcoinOutPoint { txid: htlc_success_tx.txid(), vout: 0 }));
```

Comment:
The script here is a bit confused - sometimes its the redeemscript sometimes its the script itself. Also, a script isn't *really* the best way to tell a user how to spend an output - I think an enum (that might contain data, eg a redeemscript) may be more useful.

Reply:
Hmm something like output script descriptor seems to be the right solution ? But it's still not into rust-bitcoin..

Reply:
Indeed, though I dont think the descriptor stuff is always sufficiently expressive for custom scripts. A simple enum with clear docs that explain what you have to put in the witness stack for the spend to succeed should be sufficient.